### ISSUE TITLE
v2.3

### ISSUE THREAD

**Issue by andris9** [nodemailer/nodemailer#570]
This is the tracking ticket for the next planned release of Nodemailer (v2.3.0). Expected release date is March 3rd.

  - [x] Dynamic loading of transports. #569 (*v2.3.0-beta.0*)
  - [x] Built-in support for proxy protocols (SOCKS, HTTP CONNECT) in addition to the current custom socket feature  (*v2.3.0-beta.1*)

**Comment by andris9** [comment]
Released v2.3.0-beta.1 that introduces built in proxy support. Provide a `proxy` configuration option to use it.

```javascript
var smtpConfig = {
    host: 'smtp.gmail.com',
    port: 465,
    ...,
    //proxy config
    // assumes a SOCKS5 proxy running on port 1080
    proxy: 'socks5://localhost:1080/'
};
```

Proxy type is detected from the URL protocol.

**Comment by andris9** [comment]
Tagged latest beta as RC (*v2.3.0-rc.2*) which means that stable release date is going to be March 3rd.

**Comment by andris9** [comment]
Published as v2.3.0
